include/exadg/operators/operator_base.cpp

```diff
@@ -425,6 +425,11 @@ OperatorBase<dim, Number, n_components>::add_diagonal(VectorType & diagonal) con
   {
     if(data.use_cell_based_loops)
     {
+      AssertThrow(
+        not evaluate_face_integrals() ||
```

Comment:
`evaluate_face_integrals()` will be `true` due to the if-statement a few lines above. So I believe this code is hard to understand/read.

Reply:
Principally, we decided to write `or` in ExaDG. This comment will probably be superfluous due to the other comment above suggesting to remove one of the booleans.

Reply:
I have removed the additional check.